### DISCUSSION TITLE
accessing messages from outside the handler and bookkeeping of custom messages

### DISCUSSION THREAD

**Discussion by ghost** [manzt/anywidget#384] [JavaScript]
I want to receive a message from python in js. 
I followed the example on https://anywidget.dev/en/jupyter-widgets-the-good-parts/ regarding custom messages.

```javascript
this.on("msg:custom", msg => {
      console.log("msg custom", msg)
        ret = msg}
    )
    console.log("ret:", ret)
  }
```

"this" refers to a message handler wrote for the javascript side.
In my little text example I run send() on a button click and then echo the message.
This is what gets printed to the console on the first run
```
ret: undefined 
msg custom 
Object { fn: "fun", args: (2) […] }
```
on subsequent runs the number of "msg custom" lines increases
```
ret: undefined 
msg custom 
Object { fn: "fun", args: (2) […] }

ret: undefined 
msg custom 
Object { fn: "fun", args: (2) […] }
msg custom 
Object { fn: "fun", args: (2) […] }
```

How can I access the message outside of the msg handler and why do the messages pile up. Do I need to clear some queue?

**Comment by manzt** [comment]
> How can I access the message outside of the msg handler and why do the messages pile up. Do I need to clear some queue?

Can you please provide a minimal reproducible example? You can use the example from the documentation as a starting point. There is not context here to what `ret` is originally defined or how this code is being used.

This doesn't seem to be an issue with anywidget, but rather a question about eventful programming in JavaScript. I am moving to a discussion.
